addons/web/static/tests/views/list_tests.js

```diff
@@ -8536,6 +8554,102 @@ QUnit.module('Views', {
         list.destroy();
     });
 
+    QUnit.test('editable list: multi-edit field with daterange widget', async function (assert) {
+        assert.expect(6);
+
+        const list = await createView({
+            View: ListView,
+            model: 'daterange',
+            data: this.data,
+            arch: `
+                <tree multi_edit="1">
+                    <field name="date_start" widget="daterange" options="{'related_end_date': 'date_end'}" />
+                    <field name="date_end" widget="daterange" options="{'related_start_date': 'date_start'}"/>
+                </tree>`,
+            session: {
+                getTZOffset: function () {
+                    return 330;
+                },
+            },
+        });
+
+        await testUtils.dom.click(list.$('thead .o_list_record_selector:first input'));
+        await testUtils.dom.click(list.$('.o_data_row:first .o_data_cell:eq(0) .o_field_date_range'));
+        await testUtils.dom.click(list.$('.o_data_row:first .o_data_cell:eq(0) .o_field_date_range'));
+
+        // change dates via the daterangepicker
+        await testUtils.dom.triggerMouseEvent($('.daterangepicker:first .drp-calendar.left .available:contains("16")'), 'mousedown');
+        await testUtils.dom.triggerMouseEvent($('.daterangepicker:first .drp-calendar.right .available:contains("12")'), 'mousedown');
+
+
+        const $applyBtn = $('.daterangepicker:first .applyBtn');
+        assert.ok($applyBtn.length === 1 && !$applyBtn.disabled, 'Should only have 1 apply button in the daterangepicker and this button should be enabled.');
+
+        // daterange selected
+        const daterange = $('.daterangepicker:first .drp-selected').text();
+        let [dateStart, dateEnd] = daterange.split(' - ');
+        const dateFormat = "MM/DD/YYYY";
+
+        testUtils.mock.intercept(list, "field_changed", function (ev) {
+            const changes = ev.data.changes;
+            assert.ok(changes.hasOwnProperty('date_start') && changes.hasOwnProperty('date_end'));
+            const actualDateStart = changes.date_start.format(dateFormat);
+            const actualDateEnd = changes.date_end.format(dateFormat);
+            assert.strictEqual(actualDateStart, dateStart.trim());
+            assert.strictEqual(actualDateEnd, dateEnd.trim());
+        }, true);
+
+        // Apply the changes
+        await testUtils.dom.click($applyBtn);
+
+        assert.containsOnce(document.body, '.modal', 'The confirm dialog should appear to confirm the multi edition.');
+
+        // Valid the confirm dialog
+        await testUtils.dom.click($('.modal .btn-primary'));
+
+        assert.containsNone(document.body, '.modal', 'No confirm dialog should be visible in the dom.');
+
+        list.destroy();
```

Comment:
you don't control the final state of the list, is it normal ?

Reply:
The problem is I don't have the modifiersData in the renderer and since I use it in the `_saveMultipleRecords` of list controller, I cannot find the modifiersData for the related field and thus the related field is not updated in the unit test.
I don't know how to define the all modifiersData in the unit test.